src/node_internals.h

```diff
@@ -204,6 +204,34 @@ v8::MaybeLocal<v8::Object> New(Environment* env,
 // because ArrayBufferAllocator::Free() deallocates it again with free().
 // Mixing operator new and free() is undefined behavior so don't do that.
 v8::MaybeLocal<v8::Object> New(Environment* env, char* data, size_t length);
+
+// Takes ownership of |buf|.
+template<typename T>
+static v8::MaybeLocal<v8::Object> New(Environment* env,
+                                      MaybeStackBuffer<T>* buf) {
+  v8::MaybeLocal<v8::Object> ret;
+  char* src = reinterpret_cast<char*>(buf->out());
+  const size_t len_in_bytes = buf->length() * sizeof(T);
+
+  if (buf->IsAllocated())
+    ret = New(env, src, len_in_bytes);
+  else
+    ret = Copy(env, src, len_in_bytes);
+
+  if (ret.IsEmpty())
+    return ret;
+
+  if (buf->IsAllocated())
+    buf->Release();
+
+  static_assert(sizeof(T) == 1 || sizeof(T) == 2,
+                "Currently only one- or two-byte buffers are supported");
+  if (sizeof(T) > 1 && IsBigEndian()) {
+    SPREAD_BUFFER_ARG(ret.ToLocalChecked(), retbuf);
+    SwapBytes16(retbuf_data, retbuf_length);
+  }
```

Comment:
I think I would prefer it if we did the `SwapBytes16()` at the call side and kept `Buffer::New()`s  job restricted to turning the memory block into a `Buffer` instance?